### ISSUE TITLE
Application Scenario

### ISSUE THREAD

**Issue by victorchou0811** [Lcrypto/Simulated-annealing-lifting-QC-LDPC#4]
Dear Vasiliy,
  I would like to ask if such algorithm support the QC-LDPC construction in term of a base matrix with long length? For example, I want to construct a long length LDPC code whose column number is 10^6. Firstly, I construct a base matrix whose column number is 5*10^4 according to optimized degree distribution with PEG algorithm. Thus, the size of circulant permutation matrix is 20. Or the number of column of base matrix is 1000, the size of circulant permutation matrix is 1000. Then, the exponent matrix of QC-LDPC is constructed by "Simulated-annneling-lifting-QC-LDPC" algorithm. When I'm using such algorithm to construct long QC-LDPC code, It takes a lots of time. 

Best Regards
Victor Chou

**Comment by Lcrypto** [comment]
Yes, it support long length. I construct up to 800000 code length. Soon article about long length code construction shall be publish. Time for construction depend from circulant, protograph size and cycles constraint (cycles size, EMD). Before define cycle constrains better to check balance cycle (voltage graph) which define girth Upper bound, to applied reasonable parameters on Girth and EMD. For some even large circulant QC extension with desire cycle properties Couldn't exist. Use parallel run with different seed as I describe in article. Current it SA one of the best way to construct long length QC-LDPC codes. With BR Vasiliy!

**Comment by victorchou0811** [comment]
Dear Vasiliy, thanks for your kind response. Because it is difficult to determine reasonable values of EMD and Girth in SA algorithm, I would like to ask you how to check balance cycle. Would you like to tell me the related references or paper? Thanks!!! 
Best Regards
Victor Chou

**Comment by Lcrypto** [comment]
Dear Victor, 
main articles 
C. A. Kelley and J. L. Walker, "LDPC codes from voltage graphs," 2008 IEEE International Symposium on Information Theory, Toronto, ON, 2008, pp. 792-796.
G. Xiao and Z. Nan, "Analysis of balanced cycles of QC-LDPC codes," 2011 Eighth International Conference on Wireless and Optical Communications Networks, Paris, 2011, pp. 1-5.
I shall try to found my simple stand alone balance cycles enumerating tool and upload to GitHub.
Best Regard 
Usatyuk VAsiliy
